language/tools/move-mv-llvm-compiler/src/stackless/module_context.rs

```diff
@@ -832,4 +836,214 @@ impl<'mm, 'up> ModuleContext<'mm, 'up> {
         }
         attrs
     }
+
+    fn generate_global_str_slice(&self, s: &str) -> llvm::Global {
+        let llcx = &self.llvm_cx;
+
+        // Create an LLVM global for the string.
+        let str_literal_init = llcx.const_int_array::<u8>(s.as_bytes()).as_const();
+        let str_literal = self
+            .llvm_module
+            .add_global2(str_literal_init.llvm_type(), "str_literal");
+        str_literal.set_constant();
+        str_literal.set_alignment(1);
+        str_literal.set_unnamed_addr();
+        str_literal.set_linkage(llvm::LLVMLinkage::LLVMPrivateLinkage);
+        str_literal.set_initializer(str_literal_init);
+
+        // Create an LLVM global for the slice, which is a struct with two fields:
+        // - pointer to the string literal
+        // - integer length of the string literal
+        let slice_len = s.len();
+        let slice_init = llcx.const_struct(&[
+            str_literal.ptr(),
+            llcx.const_int_array::<u8>(&slice_len.to_le_bytes())
+                .as_const(),
```

Comment:
Nit: It's a bit more natural to just use an integer constant here, rather than byte array:
`llvm::Constant::int(llcx.int_type(64), U256::from(slice_len as u128))`

Reply:
For some reason rust compiler generates the slice constants as a pair of string literal address and a byte array for slice length, but I guess it doesn't matter. I changed it to an int constant.